tests/jballerina-unit-test/src/test/resources/test-src/types/table/record-type-table-key.bal

```diff
@@ -201,6 +201,85 @@ function testVariableNameFieldAsKeyField() {
     assertEquality(expected, tb.toString());
 }
 
+function testDefaultValueFieldAsKeyField() {
+    string expected = "[{\"id\":1,\"name\":\"Jo\"},{\"id\":2,\"name\":\"Amy\"}]";
+    table<record {readonly int id = 1; string name;}> key (id) tb = table [
+        {id, name: "Jo"},
+        {id: 2, name: "Amy"}
+    ];
+    assertEquality(expected, tb.toString());
```

Comment:
Better to assert by value since we may lose type information when comparing string representations.

Reply:
Updated. Please check